TickSpec/LineParser.fs

```diff
@@ -39,49 +44,54 @@ let tryRegex input pattern =
 let startsWith (pattern:string) (s:string) =
     s.StartsWith(pattern, System.StringComparison.InvariantCultureIgnoreCase)
 
-let (|Scenario|_|) (s:string) =
-    let s = s.Trim()
-    if s |> startsWith "Scenario" || s |> startsWith "Story" then
-        Scenario s |> Some else None
-let (|IsBackground|_|) s =
+let (|FeatureLine|_|) s =
+    tryRegex s "^\s*Feature:\s(.*)"
+    |> Option.map (fun t -> FeatureLine (t.Trim()))
+let (|ScenarioLine|_|) (s:string) =
+    let trimmed = s.Trim()
+    let matches =
+        [ "Scenario"; "Story" ]
+        |> Seq.choose (fun x ->
+            if trimmed |> startsWith x then Some trimmed
+            else None)
+    if matches |> Seq.isEmpty then None
+    else ScenarioLine (matches |> Seq.head) |> Some
```

Comment:
the above seems complicated, original or the following feels simpler
```FSharp
[ "Scenario"; "Story" ]
|> Seq.tryPick (fun x -> 
    if trimmed |> startsWith x
    then ScenarioLine trimmed |> Some 
    else None) 
```

Reply:
Thanks.